src/casync.c

```diff
@@ -834,7 +835,7 @@ static int ca_sync_process_decoder_request(CaSync *s) {
                 if (r < 0)
                         return r;
 
-                if (index_size != object_size) {
+                if ((size_t) index_size != object_size) {
```

Comment:
This cast should not be necessary. Both are unsigned, and width promotions are done automatically and without warning afair.